src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -45,6 +47,10 @@ public CommandResult execute(String commandText) {
         final CommandResult result = command.execute();
         updateConfigStorage(oldConfig);
         updateTaskBookStorage(taskBookListener);
+
+        if (model.hasUncommittedChanges() && !(command instanceof UndoCommand) && !(command instanceof RedoCommand) ) {
```

Comment:
1) Right, that's handled by `hasUncommittedChanges()`.
2) Does undo/redo actually make any uncommitted changes? Will `hasUncommittedChanges()` return true after an undo/redo? After all, you are reverting back to an earlier commit, so there should not be any differences between the task book and the commit you just reverted back to.

If `hasUncommittedChanges()` is still returning true after an undo/redo, it means there is a bug.